### PR TITLE
Hy vendors api

### PR DESCRIPTION
**Pull request by HristoYonkov**
1. Create new Vendor api functionality added.
2. Edit  Vendor api functionality added.
3. Delete Vendor api functionality added.

image

### CODE REVIEW

File: src/components/features/actionsMenu/VendorTableActionsMenu.tsx [hunk 1]
```diff
@@ -3,34 +3,100 @@ import Menu from '@mui/material/Menu'
...
 
-  const actionHandler = (option: string) => {
-    specificOptionHandler(option)
+  const onDiscardClick = () => {
     handleClose()
   }
 
+  const queryClient = useQueryClient()
+
+  const mutationDelete = useMutation({
```
Comment (vexleet) [inline src/components/features/actionsMenu/VendorTableActionsMenu.tsx]: you can check out what @Hristo-Boyadzhiev has done with the zones in `hooks/services/zones` and you can move the query hooks and make them like what he has done to make things simpler and easier to read.

File: src/components/features/actionsMenu/VendorTableActionsMenu.tsx [hunk 2]
```diff
@@ -53,17 +119,39 @@ export default function VendorTableActionsMenu({
...
         {options.map((option) => (
-          <MenuItem key={option.value} onClick={() => actionHandler(option.value)}>
-            {translate(option.title)}
+          <MenuItem key={option} onClick={() => actionHandler(option)}>
+            {option}
           </MenuItem>
         ))}
       </Menu>
 
+      {selectedOption === 'Редактирай' && (
```
Comment (vexleet) [inline src/components/features/actionsMenu/VendorTableActionsMenu.tsx]: dont have options in Bulgarian. make them in English. Like what you have done with the "move entry" one

File: src/components/features/forms/NewVendorForm.tsx [hunk 3]
```diff
@@ -47,13 +62,30 @@ export default function NewVendorForm({
...
+              labelId="demo-multiple-markers-label"
+              id="demo-multiple-markers"
+              multiple
+              value={field.value || []}
+              onChange={(e) => field.onChange(e.target.value)}
+              renderValue={(selected) => (selected as string[]).join(', ')}>
+              {markers.map((marker: MarkerDto) => (
+                <MenuItem key={marker.id} value={marker.id?.toString()}>
+                  <ListItemText primary={marker.name} />
+                </MenuItem>
```
Comment (vexleet) [inline src/components/features/forms/NewVendorForm.tsx:96]: we need to show the marker name and not id when we have selected something
image

You can check what Boyadzhiev has done in Zones Form to fix that.

File: src/components/features/actionsMenu/VendorTableActionsMenu.tsx (hunk 1 above)
Comment (HristoYonkov) [reply src/components/features/actionsMenu/VendorTableActionsMenu.tsx]: It is fixed now, also i call TableActionsMenu to build dynamicly buttons for actions menu

File: src/components/features/actionsMenu/VendorTableActionsMenu.tsx (hunk 2 above)
Comment (HristoYonkov) [reply src/components/features/actionsMenu/VendorTableActionsMenu.tsx]: Fixed.

File: src/components/features/forms/NewVendorForm.tsx (hunk 3 above)
Comment (HristoYonkov) [reply src/components/features/forms/NewVendorForm.tsx:96]: Fixed.

File: src/components/features/admin/VendorsTable.tsx [hunk 4]
```diff
@@ -31,20 +47,24 @@ export default function VendorsTable() {
...
+      id: vendor.id!,
+      name: vendor.name!,
+      vendorNumber: vendor.systemNumber!,
+      markers:
+        vendor.markers!.length > 0 ? (
+          <ChipsList
+            items={vendor.markers?.map((marker) => marker.markerName!) || ([] as string[])}
+          />
+        ) : (
+          <Typography>Empty</Typography>
```
Comment (vexleet) [inline src/components/features/admin/VendorsTable.tsx]:
```suggestion
          <Typography>-</Typography>
```

**Review by vexleet — APPROVED** [review]

**Pull request merged**